packages/shared/src/components/fields/SearchField.tsx

```diff
@@ -111,21 +140,26 @@ export const SearchField = forwardRef(function SearchField(
         type={type}
         aria-describedby={describedBy}
         autoComplete="off"
-        className="flex-1"
+        className={classNames(
+          'flex-1',
+          getInputFontColor({ readOnly, disabled, hasInput, focused }),
+        )}
         required
       />
-      {rightChildren || (
-        <button
-          type="button"
+      {((hasInput && isPrimary) || isSecondary) && (
+        <Button
+          {...rightButtonProps}
+          className={isSecondary ? 'btn-primary' : 'btn-tertiary'}
+          buttonSize="xsmall"
           title="Clear query"
```

Comment:
Should we not make this optional for the future?

Reply:
Ah, yes. True, we should.